lib/praxis/types/multipart_array.rb

```diff
@@ -51,9 +51,7 @@ def self.name_type(type=nil)
         @name_type = Attributor.resolve_type type
       end
 
-      def self.payload_type(type=nil, **opts, &block)
-        return @payload_type if type.nil?
-
+      def self.payload_type(type=Attributor::Struct, **opts, &block)
```

Comment:
Let's leave the `type=nil` default here, and the `return ... if type.nil?`, so we can keep this as a reader too.

Then you can handle the `Struct` default similarly to how we do in Attributor here https://github.com/rightscale/attributor/blob/master/lib/attributor/dsl_compiler.rb#L101-L102 , with something akin to:
```ruby
if type.nil?
  if block_given?
    type = Attributor::Struct
...
```